### DISCUSSION TITLE
how do I parse unknown json data format?

### DISCUSSION THREAD

**Discussion by refaqtor** [vlang/v#18565] [Questions and Answers] [answered]
I'm trying to read JSON of which I don't have defined struct to load them into as called for by the current std lib 
```
fn decode(typ voidptr, s string) !voidptr
```

Am I misunderstanding how that works?  How does one do the, seemingly, simple task of reading in unknown JSON data format?

**Comment by JalonSolov** [comment, marked as answer]
Use functions from `x.json2`, rather than regular `json` module.  The plan is for `x.json2` to replace `json` once it is complete.  For now, you may wind up using some functions from both to get complete JSON support.

**Reply by refaqtor** [reply]
ah! thank you.  raw_decode    and    as_map   are the key for me.

**Reply by einar-hjortdal** [reply]
However note that json2.decode only seems to decode string values
